src/minion.ts

```diff
@@ -299,17 +299,22 @@ export default class Minion {
     resolve: (value?: any) => void,
     reject: (reason?: any) => void
   ) {
-    const info = (await this.backend.listJobs(0, 1, {ids: [id]})).jobs[0];
-    if (info === undefined) {
-      resolve(null);
-    } else if (info.state === 'finished') {
-      resolve(info);
-    } else if (info.state === 'failed') {
-      reject(info);
-    } else if (signal !== null && signal.aborted === true) {
-      reject(new AbortError());
-    } else {
-      setTimeout(() => this._result(id, interval, signal, resolve, reject), interval);
+    const rerun = () => this._result(id, interval, signal, resolve, reject);
+    try {
+      const info = (await this.backend.listJobs(0, 1, {ids: [id]})).jobs[0];
+      if (info === undefined) {
+        resolve(null);
+      } else if (info.state === 'finished') {
+        resolve(info);
+      } else if (info.state === 'failed') {
+        reject(info);
+      } else if (signal !== null && signal.aborted === true) {
+        reject(new AbortError());
+      } else {
+        setTimeout(rerun, interval);
+      }
+    } catch (error: any) {
+      setTimeout(rerun, interval);
```

Comment:
Wouldn't a single `finally` block solve this with less code? 🤔

Reply:
A single `finally` block would be executed every time after we've enter the `try` block, even if we called `resolve` or `reject` (in those cases, we don't need another invocation of `_result`).

Reply:
If you don't like the length `try` block, it could also be implemented like this, at the expense of using `let` instead of `const`:

```ts
let info: JobInfo;
try {
  info = (await this.backend.listJobs(0, 1, {ids: [id]})).jobs[0];
} catch (error: any) {
  setTimeout(rerun, interval);
  return;
}

if (info === undefined) {
...
```

There would be a few alternative code flows to avoid duplicating the `setTimeout` command, but I'd consider each of them less readable.